### PR TITLE
plugin/rewrite: add response rewrite docs

### PR DESCRIPTION
**Pull request by greenpau**
### 1. What does this pull request do?

Add DNS response re-write documentation for `plugin/rewrite`.

### 2. Which issues (if any) are related?

See: https://github.com/coredns/coredns/issues/1313#issuecomment-358983475

### 3. Which documentation changes (if any) need to be made?

The `plugin/rewrite` documentation.

### CODE REVIEW

**Comment by greenpau** [comment]
@johnbelamaric , per your request, please review.

File: plugin/rewrite/README.md [hunk 1]
```diff
@@ -141,3 +141,54 @@ Thus:
 
 * Incoming Request Name: `ftp-us-west-1.example.org`
 * Re-written Request Name: `ftp.service.us-west-1.consul`
+
+### Response Rewrites
+
+When re-writing incoming DNS requests' names, `coredns` re-write the `QUESTION SECTION`
```
Comment (johnbelamaric) [inline plugin/rewrite/README.md]: s/`coredns`/CoreDNS/

s/re-write/re-writes/

File: plugin/rewrite/README.md [hunk 2]
```diff
@@ -141,3 +141,54 @@ Thus:
...
+section of the requests. It may be necessary to re-write the `ANSWER SECTION` of the
+requests, because some DNS resolvers would treat the mismatch between `QUESTION SECTION`
+and `ANSWER SECTION` as a man-in-the-middle attack (MITM).
+
+For example, a user tries to resolve `ftp-us-west-1.coredns.rocks`. The
+`coredns` configuration file has the following rule:
```
Comment (johnbelamaric) [inline plugin/rewrite/README.md]: s/`coredns`/CoreDNS/

File: plugin/rewrite/README.md [hunk 3]
```diff
@@ -141,3 +141,54 @@ Thus:
...
+
+```
+rewrite name regex (.*)-(us-west-1)\.coredns\.rocks {1}.service.{2}.consul
+```
+
+`coredns` instance re-wrote the request to `ftp-us-west-1.coredns.rocks` with
```
Comment (johnbelamaric) [inline plugin/rewrite/README.md]: ditto

**Review by johnbelamaric — CHANGES_REQUESTED** [review]
Looks pretty good. Some nits. Also:

1. Add a general syntax for the long form/answer rules. I think the way the parser works you can put multiple rules inside the braces.

2. Move the Name Rewrites section above the EDNS0 section, since I think it will be more common.

**Comment by greenpau** [comment]
> Add a general syntax for the long form/answer rules.

@johnbelamaric , will do.

> I think the way the parser works you can put multiple rules inside the braces.

@johnbelamaric , I don't think so. One rule per curly brace.

**Comment by greenpau** [comment]
> Move the Name Rewrites section above the EDNS0 section, since I think it will be more common.

@johnbelamaric , ack!

**Comment by greenpau** [comment]
@johnbelamaric , please review once again.

**Review by johnbelamaric — APPROVED** [review]

**Pull request merged**